include/unifex/tag_invoke.hpp

```diff
@@ -32,6 +32,21 @@ struct tag_invoke_cpo {
     return tag_invoke((CPO&&)cpo, (Args &&) args...);
   }
 };
+
+template<typename CPO, typename... Args>
+using tag_invoke_result_t = decltype(tag_invoke(static_cast<CPO&&(*)()>(nullptr)(), static_cast<Args&&(*)()>(nullptr)()...));
+
+struct yes_type { char dummy; };
+struct no_type { char dummy[2]; };
+
+template<typename CPO, typename... Args>
+auto try_tag_invoke(int)
+  noexcept(noexcept(tag_invoke(static_cast<CPO&&(*)() noexcept>(nullptr)(), static_cast<Args&&(*)() noexcept>(nullptr)()...)))
+  -> decltype(tag_invoke(static_cast<CPO&&(*)() noexcept>(nullptr)(), static_cast<Args&&(*)() noexcept>(nullptr)()...), yes_type{});
```

Comment:
Why not reuse `tag_invoke_result_t` in the return type here? Also, guard against overloaded `operator,` by putting `void(), ` before `yes_type{}` here.

Reply:
I was trying to use the same `noexcept`-qualified `static_cast` expressions in both the noexcept and return-type components - thinking that the compiler might be able to reuse the overload resolution results for both expressions.

If it doesn't make a difference then, yes, I may as well use `tag_invoke_result_t`.